File: memento/java/TextUndoCareTaker.java
package Memento;

import java.util.Stack;

// Memento'ların referansının tutulduğu yerdir.
// UML diyagramındaki CareTaker yapısına denk gelmektedir.
class TextUndoCareTaker {
    private Stack<TextMemento> _mementos;

    public TextUndoCareTaker() {
        _mementos = new Stack<TextMemento>();
    }

    // Çağrılma işlemi yapıldığında yığının en üstündeki Memento örneği silinir ve geriye döndürülür.
    // Ekleme işlemi yapıldığında yığının en üstüne Memento örneği eklenir.
    // Klasik Stack.

    public void setMemento(TextMemento memento) {
        _mementos.push(memento);
    }

    public TextMemento getMemento() {
        return _mementos.pop();
    }
}

File: memento/go/text-undo-care-taker.go
package main

// TextUndoCareTaker Memento'ların referansının tutulduğu yerdir.
// UML diyagramındaki CareTaker yapısına denk gelmektedir.
type TextUndoCareTaker struct {
	mementos *Stack
}

func NewTextUndoCareTaker() *TextUndoCareTaker {
	return &TextUndoCareTaker{
		mementos: &Stack{},
	}
}

// SetMemento Çağrılma işlemi yapıldığında yığının en üstündeki Memento örneği silinir ve geriye döndürülür.
// Ekleme işlemi yapıldığında yığının en üstüne Memento örneği eklenir.
// Klasik Stack.
func (t *TextUndoCareTaker) SetMemento(memento *TextMemento) {
	t.mementos.Push(memento)
}

func (t *TextUndoCareTaker) GetMemento() *TextMemento {
	return t.mementos.Pop()
}
